### PR TITLE
add Jupyter Sphinx support

### LINKED ISSUE #35: compatibility with jupyter-sphinx
I would like to use this on `rst` files that are used in combination with https://github.com/jupyter/jupyter-sphinx.

Here the code blocks are formatted like:
```rst
.. jupyter-execute::
    :hide-code:

    code_here

# or

.. jupyter-execute::

    code_here
```

Specifically, I would like to use it for the Adaptive [documentation](https://adaptive.readthedocs.io/en/latest/tutorial/tutorial.Learner1D.html) ([repo](https://github.com/python-adaptive/adaptive/)).

I've made an attempt myself, but I can't match @asottile's regex wizardry 😉

### PR DESCRIPTION
**Pull request by basnijholt**
Closes #35.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,11 @@ Following additional parameters can be used:
 
         def hello():
             print("hello world")
+
+    .. jupyter-execute::
+        :hide-code:
+
+        print("This is for jupyter-sphinx")
```
Comment (asottile) [inline README.md]: don't need this example here -- these are just meant to explain the different file types (you can rebase out this commit -- thanks!)

File: tests/blacken_docs_test.py [hunk 2]
```diff
@@ -345,3 +345,47 @@ def test_integration_ignored_syntax_error(tmpdir, capsys):
         'f(\n'
         '```\n'
     )
+
+def test_format_src_rst_jupyter_sphinx():
```
Comment (asottile) [inline tests/blacken_docs_test.py]: hehe looks like you didn't run the pre-commit hooks

File: blacken_docs.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 PY_LANGS = '(python|py|sage|python3|py3|numpy)'
 RST_RE = re.compile(
     rf'(?P<before>'
-    rf'^(?P<indent> *)\.\. (code|code-block|sourcecode|ipython):: {PY_LANGS}\n'
+    rf'^(?P<indent> *)\.\. (jupyter-execute::|(code|code-block|sourcecode|ipython):: {PY_LANGS})\n'
```
Comment (asottile) [inline blacken_docs.py]: looks like this line ends up being too long as well -- I might split this up by putting above:

```python
BLOCK_TYPES = '(code|code-block|sourcecode|ipython)'
```
and then swapping that part out for `{BLOCK_TYPES}` inside the regex

:+1:

Comment (basnijholt) [reply blacken_docs.py]: Done!

File: tests/blacken_docs_test.py (hunk 2 above)
Comment (basnijholt) [reply tests/blacken_docs_test.py]: Yes, when I run them it changes A LOT of code, also a lot that I didn't touch.

I am not sure what I am doing wrong...

Comment (asottile) [reply tests/blacken_docs_test.py]: huh!  can you show some diffs, that's kind of surprising!

Comment (basnijholt) [reply tests/blacken_docs_test.py]: Hmmm, I cannot reproduce.

When I ran `pre-commit run --all` before, it also ran `black`.

Something must have been in an incorrect state.

File: blacken_docs.py [hunk 4]
```diff
@@ -20,9 +20,10 @@
     re.DOTALL | re.MULTILINE,
 )
 PY_LANGS = '(python|py|sage|python3|py3|numpy)'
+BLOCK_TYPES = '(code|code-block|sourcecode|ipython)'
 RST_RE = re.compile(
     rf'(?P<before>'
-    rf'^(?P<indent> *)\.\. (code|code-block|sourcecode|ipython):: {PY_LANGS}\n'
+    rf'^(?P<indent> *)\.\. (jupyter-execute::|(BLOCK_TYPES):: {PY_LANGS})\n'
```
Comment (asottile) [inline blacken_docs.py]: this needs to be curly braces so it substitutes the string above

Comment (basnijholt) [reply blacken_docs.py]: Oops, sorry 😅

File: blacken_docs.py [hunk 5]
```diff
@@ -20,9 +20,10 @@
     re.DOTALL | re.MULTILINE,
 )
 PY_LANGS = '(python|py|sage|python3|py3|numpy)'
+BLOCK_TYPES = '(code|code-block|sourcecode|ipython)'
 RST_RE = re.compile(
     rf'(?P<before>'
-    rf'^(?P<indent> *)\.\. (code|code-block|sourcecode|ipython):: {PY_LANGS}\n'
+    rf'^(?P<indent> *)\.\. (jupyter-execute::|{BLOCK_TYPES}):: {PY_LANGS})\n'
```
Comment (asottile) [inline blacken_docs.py]: still one extra paren here :D
```suggestion
    rf'^(?P<indent> *)\.\. (jupyter-execute::|{BLOCK_TYPES}:: {PY_LANGS})\n'
```

**Comment by asottile** [comment]
released in v1.5.0

**Pull request merged**